### PR TITLE
Refactor/ruff lint

### PR DESCRIPTION
**Pull request by redur**
Introduction of Ruff. A very fast code formatter and code linter. Settings are compatible with previous isort and black settings. But now, more rules are added.

The PR comes with refactoring of the code to be ruff compliant.

Code runs, metrics are not affected.

### CODE REVIEW

File: src/stratigraphy/benchmark/score.py [hunk 1]
```diff
@@ -77,7 +87,8 @@ def evaluate_matching(predictions_path: Path, ground_truth_path: Path) -> tuple[
         overall_recall = sum(document_level_metrics["recall"]) / len(document_level_metrics["recall"])
         logging.info("Macro avg:")
         logging.info(
-            f"F1: {f1(overall_precision, overall_recall):.1%}, precision: {overall_precision:.1%}, recall: {overall_recall:.1%}"
+            f"F1: {f1(overall_precision, overall_recall):.1%},"
+            "precision: {overall_precision:.1%}, recall: {overall_recall:.1%}"
```
Comment (stijnvermeeren-swisstopo) [inline src/stratigraphy/benchmark/score.py:91]:
```suggestion
            f"precision: {overall_precision:.1%}, recall: {overall_recall:.1%}"
```

**Comment by redur** [comment]
Merge conflicts resolved. Tested & metrics still remain the same compared to branch feat/indicator_lines_v2

**Review by stijnvermeeren-swisstopo — CHANGES_REQUESTED** [review]
Please also update the pre-commit section in the README.

So for this pre-commit hook to run, it is not necessary to install ruff in the Conda environment? However, if I also want to run ruff in my IDE (PyCharm), then probably I do need to explicitly install it as a dependency, or how does that work?

**Comment by redur** [comment]
> Please also update the pre-commit section in the README.
> 
> So for this pre-commit hook to run, it is not necessary to install ruff in the Conda environment? However, if I also want to run ruff in my IDE (PyCharm), then probably I do need to explicitly install it as a dependency, or how does that work?

You do not need to install ruff in your conda environment. I believe that the source code of ruff is downloaded from the github repo by pre-commit. And pre-commit calls that code inside our environment. Therefore we don't need to install any of the code. I may be mistaken on that one though. But for sure: No, you don't need to install ruff in order to use it with pre-commit.

Your next question: I am not familiar with how pycharm does the linting. I believe you would need to install ruff in our dev environment and select the right interpreter in pycharm for it to work. Maybe pycharm offers a similar functionality as pre-commit, and you can specify the source code directly. I don't know about that.

File: src/stratigraphy/benchmark/score.py (hunk 1 above)
Comment (redur) [reply src/stratigraphy/benchmark/score.py]: Thanks.

**Review by stijnvermeeren-swisstopo — APPROVED** [review]

**Pull request merged**